### PR TITLE
Ng15 patch

### PR DESCRIPTION
**Pull request by aftab10662**
ng15 patch.

### CODE REVIEW

**Comment by aftab10662** [comment]
@Rick-Anderson Request you to review the changes.

**Comment by Rick-Anderson** [comment]
@spboyer please review

**Comment by Rick-Anderson** [comment]
@aftab10662  you have a build error
Warning, treated as error:
/home/travis/build/aspnet/Docs/aspnet/client-side/angular.rst:290: WARNING: Include file '/home/travis/build/aspnet/Docs/aspnet/client-side/angular/sample/AngularSample/src/AngularSample/Views/Home/Component.cshtml' not found or reading it failed

Travis quits on the first build error. Are you able to build the doc locally and test?

**Comment by aftab10662** [comment]
@Rick-Anderson, yes i am able to build it successfully using VS2015 Community. 
1>  Build succeeded.
1>           0 Warning(s).
1>           0 Error(s).

And Component.cshtml is not getting used anywhere.
It is Components.cshtml which is in use.
Not getting clue.

**Comment by aftab10662** [comment]
@Rick-Anderson i found the issue.
 I corrected and committed to branch.it is available in PR

**Comment by Rick-Anderson** [comment]
@spboyer please review

File: aspnet/client-side/angular/sample/AngularSample/src/AngularSample/Controllers/HomeController.cs [hunk 1]
```diff
@@ -65,5 +65,13 @@ public IActionResult Controllers()
...
         }
+		public IActionResult Components()
+		{
+			return View();
+		}
+		public IActionResult PersonComponent()
+		{
+			return View();
+		}
     }
```
Comment (spboyer) [inline aspnet/client-side/angular/sample/AngularSample/src/AngularSample/Controllers/HomeController.cs]: Can we get the tabbing fixed?

File: aspnet/client-side/angular.rst [hunk 2]
```diff
@@ -273,6 +273,35 @@ The page shows "Mary" and "Jane" that correspond to the ``firstName`` and ``last
 
 .. image:: angular/_static/controllers.png
 
+Components
+^^^^^^^^^^
+`Components <https://docs.angularjs.org/guide/component>`_ in AngularJS  are commonly used for creating custom HTML elements just like a directive. Angular developers have been using controllers and directives to build out the features for many years. Angular team introduced new component programming model in Angular 1.5. Components are the best way to program with Angular. Components offer the best of both the world when it comes to the feature of controller and directive. And writing components will make easier to migrate your application to Angular 2.
```
Comment (spboyer) [inline aspnet/client-side/angular.rst]: I would suggest re-wording the intro:

Components in Angular 1.5.x allow for the encapsulation and capability of creating individual HTML elements. In Angular 1.4.x you could achieve the same feature using the `.directive()` method.

By using the `.component()` method, development is simplified gaining the functionality of the directive and the controller. Other benefits include; scope isolation, best practices are inherent, and migration to Angular 2 becomes an easier task.

**Comment by spboyer** [comment]
@Rick-Anderson @aftab10662 - see notes

**Comment by aftab10662** [comment]
@spboyer @Rick-Anderson Thanks for reviewing the PR.

I corrected. Request you to re look. 
If anything needed from my end please let me know.

**Comment by Rick-Anderson** [comment]
@aftab10662  can you sign the cla?

**Comment by aftab10662** [comment]
@Rick-Anderson @spboyer CLA signed.

**Comment by spboyer** [comment]
:shipit:

**Pull request merged**